AlphaWallet/Core/Helpers/FirebaseReportService.swift

```diff
@@ -55,4 +55,67 @@ extension Crashlytics {
 
         setCustomKeysAndValues(keysAndValues)
     }
+
+    /// Logs large nft asset jsons to Realm, that cause crashes
+    /// - actions - update nft token balance operations
+    /// - fileSizeThreshold - max json string size, that will trigger logging
+    @discardableResult func logLargeNftJsonFiles(for actions: [AddOrUpdateTokenAction], fileSizeThreshold: Double = 10) -> Bool {
+        func logNftJsonFileIfNeeded(address: AlphaWallet.Address, server: RPCServer, balance: NonFungibleBalance) -> Bool {
+            switch balance {
+            case .assets(let rawAssets):
+                let rawAssets = rawAssets.compactMap { asset in asset.json.data(using: .utf8).flatMap { return (data: $0, value: asset) } }
```

Comment:
We can avoid unnecessary computation (String -> Data) be it big or small.

Maybe just `rawAssets.map { $0.json.count }`?

And change the threshold, if necessary to compensate?